testkit/src/main/scala/org/bitcoins/eclair/rpc/EclairRpcTestUtil.scala

```diff
@@ -215,8 +226,100 @@ trait EclairRpcTestUtil extends BitcoinSLogger {
     ()
   }
 
+  private def createNodeLink(
+      bitcoindRpcClient: Option[BitcoindRpcClient],
+      channelAmount: MilliSatoshis)(implicit actorSystem: ActorSystem): (
+      EclairRpcClient,
+      EclairRpcClient,
+      EclairRpcClient,
+      EclairRpcClient) = {
+    implicit val ec: ExecutionContext = actorSystem.dispatcher
+    val internalBitcoind = bitcoindRpcClient.getOrElse(
+      BitcoindRpcTestUtil.startedBitcoindRpcClient()
+    )
+
+    val (first, second) = createNodePair(Some(internalBitcoind))
+    val (third, fourth) = createNodePair(Some(internalBitcoind))
+
+    def open(
+        c1: EclairRpcClient,
+        c2: EclairRpcClient): Future[FundedChannelId] = {
+      openChannel(c1,
+                  c2,
+                  channelAmount.toSatoshis,
+                  MilliSatoshis(channelAmount.toLong / 2))
+    }
+
+    EclairRpcTestUtil.connectLNNodes(second, third)
+
+    val openF = Future.sequence(
+      List(open(first, second), open(second, third), open(third, fourth)))
+
+    AsyncUtil.awaitCondition(() => openF.isCompleted)
+
+    internalBitcoind.generate(3)
+
+    (first, second, third, fourth)
+  }
+
+  /**
+    * $nodeLinkDoc
+    * @note Blocks the current thread
+    * @return A 4-tuple of the created nodes' respective
+    *         [[org.bitcoins.eclair.rpc.client.EclairRpcClient EclairRpcClient]]
+    */
+  def createNodeLink(
+      bitcoindRpcClient: BitcoindRpcClient,
+  )(implicit actorSystem: ActorSystem): (
```

Comment:
Perhaps make a case class instead of this 4 tuple?

```scala
case class EclairNodes4(n1: EclairRpcClient, n2: EclairRpcClient, n3: EclairRpcClient, n4: EclairRpcClient)
```

This also allows us to add helper methods to the case class if we need to in the future

Reply:
It might even be worth genericizing this to be `n` nodes